github.com/cockroachdb/cockroach/pkg/storage: TestRaftSSTableSideloadingUpdatedReplicaID failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/39f0393d80cb62d62ce20ec1e5eccbe366f9c759

Parameters:
```
TAGS=deadlock
GOFLAGS=
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=724098&tab=buildLog

```

I180618 08:35:33.328848 800376 storage/store.go:1454  [s1] [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I180618 08:35:33.336826 800376 storage/replica_sideload_test.go:1040  olddir is /go/src/github.com/cockroachdb/cockroach/artifacts/TestRaftSSTableSideloadingUpdatedReplicaID786800310/auxiliary/sideloading/1/1.1, newdir is /go/src/github.com/cockroachdb/cockroach/artifacts/TestRaftSSTableSideloadingUpdatedReplicaID786800310/auxiliary/sideloading/1/1.2
I180618 08:35:33.336899 800376 storage/engine/rocksdb.go:699  closing rocksdb instance at "/go/src/github.com/cockroachdb/cockroach/artifacts/TestRaftSSTableSideloadingUpdatedReplicaID786800310"
fatal error: unexpected signal during runtime execution
[signal SIGSEGV: segmentation violation code=0x1 addr=0x4f9b0 pc=0x1e06dd5]

runtime stack:
runtime.throw(0x243f5fc, 0x2a)
	/usr/local/go/src/runtime/panic.go:619 +0x81
runtime.sigpanic()
	/usr/local/go/src/runtime/signal_unix.go:372 +0x28e

goroutine 800880 [syscall]:
runtime.cgocall(0x1d02480, 0xc422ef7b30, 0x3)
	/usr/local/go/src/runtime/cgocall.go:128 +0x64 fp=0xc422ef7b00 sp=0xc422ef7ac8 pc=0x8043f4
github.com/cockroachdb/cockroach/pkg/storage/engine._Cfunc_DBIterDestroy(0x2b30fb00d1e0)
	_cgo_gotypes.go:639 +0x41 fp=0xc422ef7b30 sp=0xc422ef7b00 pc=0x121ae81
github.com/cockroachdb/cockroach/pkg/storage/engine.(*rocksDBIterator).destroy.func1(0x2b30fb00d1e0)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/engine/rocksdb.go:1920 +0x56 fp=0xc422ef7b68 sp=0xc422ef7b30 pc=0x1230706
github.com/cockroachdb/cockroach/pkg/storage/engine.(*rocksDBIterator).destroy(0xc4203231f0)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/engine/rocksdb.go:1920 +0x2f fp=0xc422ef7b88 sp=0xc422ef7b68 pc=0x122651f
github.com/cockroachdb/cockroach/pkg/storage/engine.(*rocksDBIterator).Close(0xc4203231f0)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/engine/rocksdb.go:1935 +0x2b fp=0xc422ef7bb0 sp=0xc422ef7b88 pc=0x122661b
github.com/cockroachdb/cockroach/pkg/storage/engine.dbIterate(0x2b30ef6c80f0, 0x28e7260, 0xc422f05380, 0xc42003e39a, 0x6, 0x6, 0x0, 0x0, 0xc42003e3a0, 0x6, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/engine/rocksdb.go:2490 +0x36e fp=0xc422ef7ca8 sp=0xc422ef7bb0 pc=0x122b9be
github.com/cockroachdb/cockroach/pkg/storage/engine.(*RocksDB).Iterate(0xc422f05380, 0xc42003e39a, 0x6, 0x6, 0x0, 0x0, 0xc42003e3a0, 0x6, 0x6, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/engine/rocksdb.go:798 +0xbf fp=0xc422ef7d38 sp=0xc422ef7ca8 pc=0x121fcff
github.com/cockroachdb/cockroach/pkg/storage/compactor.(*Compactor).examineQueue(0xc421e26600, 0x28d4ee0, 0xc422df5500, 0x0, 0x0, 0xc420524c00)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/compactor/compactor.go:420 +0x18b fp=0xc422ef7e10 sp=0xc422ef7d38 pc=0x1356acb
github.com/cockroachdb/cockroach/pkg/storage/compactor.(*Compactor).Start.func1(0x28d4ee0, 0xc422df5500)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/compactor/compactor.go:131 +0x328 fp=0xc422ef7f88 sp=0xc422ef7e10 pc=0x1357d28
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc420389b40, 0xc422745170, 0xc422327d00)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:193 +0xe9 fp=0xc422ef7fc8 sp=0xc422ef7f88 pc=0xf96bd9
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:2361 +0x1 fp=0xc422ef7fd0 sp=0xc422ef7fc8 pc=0x85d9a1
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:186 +0xad

goroutine 1 [chan receive]:
testing.(*T).Run(0xc4209fae10, 0x243ce48, 0x2a, 0x2520e90, 0x890c01)
	/usr/local/go/src/testing/testing.go:825 +0x301
testing.runTests.func1(0xc4202880f0)
	/usr/local/go/src/testing/testing.go:1063 +0x64
testing.tRunner(0xc4202880f0, 0xc4202d3b90)
	/usr/local/go/src/testing/testing.go:777 +0xd0
testing.runTests(0xc420150980, 0x36c8940, 0x1f4, 0x1f4, 0x21c0d20)
	/usr/local/go/src/testing/testing.go:1061 +0x2c4
testing.(*M).Run(0xc420276200, 0x0)
	/usr/local/go/src/testing/testing.go:978 +0x171
github.com/cockroachdb/cockroach/pkg/storage_test.TestMain(0xc420276200)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/main_test.go:57 +0x1e3
main.main()
	_testmain.go:1056 +0x151

goroutine 7 [syscall]:
os/signal.signal_recv(0xc4200697c0)
	/usr/local/go/src/runtime/sigqueue.go:139 +0xa6
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:22 +0x22
created by os/signal.init.0
	/usr/local/go/src/os/signal/signal_unix.go:28 +0x41

goroutine 32 [chan receive]:
github.com/cockroachdb/cockroach/pkg/util/log.flushDaemon()
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:1158 +0xf1
created by github.com/cockroachdb/cockroach/pkg/util/log.init.0
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:591 +0x110

goroutine 33 [chan receive]:
github.com/cockroachdb/cockroach/pkg/util/log.signalFlusher()
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:598 +0xab
created by github.com/cockroachdb/cockroach/pkg/util/log.init.0
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:592 +0x128

goroutine 37 [select, locked to thread]:
runtime.gopark(0x2524208, 0x0, 0x23ee658, 0x6, 0x18, 0x1)
	/usr/local/go/src/runtime/proc.go:291 +0x11a
runtime.selectgo(0xc420486750, 0xc4200b5560)
	/usr/local/go/src/runtime/select.go:392 +0xe50
runtime.ensureSigM.func1()
	/usr/local/go/src/runtime/signal_unix.go:549 +0x1f4
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:2361 +0x1

goroutine 800376 [semacquire]:
sync.runtime_Semacquire(0xc42274519c)
	/usr/local/go/src/runtime/sema.go:56 +0x39
sync.(*WaitGroup).Wait(0xc422745190)
	/usr/local/go/src/sync/waitgroup.go:129 +0x72
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Stop(0xc422745170, 0x28d4e60, 0xc4200ca058)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:418 +0x229
github.com/cockroachdb/cockroach/pkg/storage.TestRaftSSTableSideloadingUpdatedReplicaID(0xc4209fae10)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica_sideload_test.go:1048 +0x6a3
testing.tRunner(0xc4209fae10, 0x2520e90)
	/usr/local/go/src/testing/testing.go:777 +0xd0
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:824 +0x2e0


ERROR: exit status 2

context canceled
make: *** [stress] Error 1
```

--- a/pkg/storage/replica_sideload_test.go
+++ b/pkg/storage/replica_sideload_test.go
@@ -991,15 +991,15 @@ func TestRaftSSTableSideloadingTruncation(t *testing.T) {
 func TestRaftSSTableSideloadingUpdatedReplicaID(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 
-	tc := testContext{}
-	stopper := stop.NewStopper()
-	defer stopper.Stop(context.TODO())
-
 	cleanup, cache, eng := newRocksDB(t)
-	tc.engine = eng
 	defer cleanup()
 	defer cache.Release()
 	defer eng.Close()
+
+	tc := testContext{}
+	stopper := stop.NewStopper()
+	defer stopper.Stop(context.TODO())
+	tc.engine = eng
 
 	tc.Start(t, stopper)
 	repl := tc.repl

